tap_postgres/client.py

```diff
@@ -275,3 +362,197 @@ def get_records(self, context: Optional[dict]) -> Iterable[Dict[str, Any]]:
         with self.connector._connect() as con:
             for row in con.execute(query):
                 yield dict(row)
+
+
+class PostgresLogBasedStream(SQLStream):
+    """Stream class for Postgres log-based streams."""
+
+    connector_class = PostgresConnector
+
+    # JSONB Objects won't be selected without type_confomance_level to ROOT_ONLY
+    TYPE_CONFORMANCE_LEVEL = TypeConformanceLevel.ROOT_ONLY
+
+    replication_key = "_sdc_lsn"
+
+    @property
+    def config(self) -> Mapping[str, Any]:
+        """Return a read-only config dictionary."""
+        return MappingProxyType(self._config)
+
+    @cached_property
+    def schema(self) -> dict:
+        """Override schema for log-based replication adding _sdc columns."""
+        schema_dict = typing.cast(dict, self._singer_catalog_entry.schema.to_dict())
+        schema_dict["properties"].update({"_sdc_deleted_at": {"type": ["string"]}})
+        schema_dict["properties"].update({"_sdc_lsn": {"type": ["integer"]}})
+        return schema_dict
+
+    def _increment_stream_state(
```

Comment:
Could you make an issue in the SDK for this. It seems like https://github.com/meltano/sdk/blob/main/singer_sdk/streams/core.py#L746 could just be swapped to something like

` if latest_record and self.replication_method in (REPLICATION_INCREMENTAL, REPLICATION_LOG_BASED)`